source/extensions/filters/http/buffer/buffer_filter.cc

```diff
@@ -70,12 +71,20 @@ Http::FilterHeadersStatus BufferFilter::decodeHeaders(Http::HeaderMap&, bool end
   }
 
   callbacks_->setDecoderBufferLimit(settings_->maxRequestBytes());
+  request_headers_ = &headers;
 
   return Http::FilterHeadersStatus::StopIteration;
 }
 
-Http::FilterDataStatus BufferFilter::decodeData(Buffer::Instance&, bool end_stream) {
+Http::FilterDataStatus BufferFilter::decodeData(Buffer::Instance& data, bool end_stream) {
+  content_length_ += data.length();
   if (end_stream || settings_->disabled()) {
+    if (request_headers_ != nullptr && request_headers_->ContentLength() == nullptr) {
```

Comment:
I see that request_headers_ being non-null is guarded by disabled so the if statement above is OK, but can you add a comment about this? It's a little non-obvious. You could potentially also ASSERT(!settings_->disabled())